src/workerd/io/compatibility-date.c++

```diff
@@ -196,14 +203,19 @@ void compileCompatibilityFlags(kj::StringPtr compatDate, capnp::List<capnp::Text
       // it redundant, because at a future date it won't be redundant, and someone could want to
       // set the flag early to make sure they don't forget later.
     }
-    if (dateValidation == CompatibilityDateValidation::CODE_VERSION &&
-        enableByFlag && !enableByDate && enableDate == nullptr) {
-      errorReporter.addError(kj::str(
-          "The compatibility flag ", enableFlagName, " is experimental and may break or be "
+    bool invalidExperiment = ((enableByFlag && !enableByDate && enableDate == nullptr) ||
+                              isExperimental) && !allowExperimentalFeatures;
+    if (dateValidation == CompatibilityDateValidation::CODE_VERSION && invalidExperiment) {
+      errorReporter.addError(kj::str("The compatibility flag ", enableFlagName, " is experimental and may break or be "
           "removed in a future version of workerd. To use this flag, you must pass --experimental "
           "on the command line."));
     }
 
+    if (dateValidation == CompatibilityDateValidation::CURRENT_DATE_FOR_CLOUDFLARE && invalidExperiment) {
+      errorReporter.addError(kj::str("The compatibility flag ", enableFlagName,
+      " is experimental and cannot yet be used in Workers deployed to Cloudflare."));
+    }
+
```

Comment:
* The experimental check should be enforced for all enum values, not just these two.
* The old logic that treated all flags without dates as experimental should go away. We only care about the `$experimental` flag now.

So it should be like this:

```suggestion
    if (isExperimental && !allowExperimentalFeatures) {
      if (dateValidation == CompatibilityDateValidation::CURRENT_DATE_FOR_CLOUDFLARE) {
        errorReporter.addError(kj::str("The compatibility flag ", enableFlagName,
            " is experimental and cannot yet be used in Workers deployed to Cloudflare."));
      } else {
        errorReporter.addError(kj::str(
            "The compatibility flag ", enableFlagName, " is experimental and may break or be "
            "removed in a future version of workerd. To use this flag, you must pass --experimental "
            "on the command line."));
      }
    }

```

Reply:
Done